kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/SchemaManager.java

```diff
@@ -70,9 +162,27 @@ public SchemaManager(
    * @param topic The Kafka topic used to determine the schema.
    */
   public void createTable(TableId table, String topic) {
-    Schema kafkaValueSchema = schemaRetriever.retrieveSchema(table, topic, KafkaSchemaRecordType.VALUE);
-    Schema kafkaKeySchema = kafkaKeyFieldName.isPresent() ? schemaRetriever.retrieveSchema(table, topic, KafkaSchemaRecordType.KEY) : null;
-    bigQuery.create(constructTableInfo(table, kafkaKeySchema, kafkaValueSchema));
+    synchronized (lock(tableCreateLocks, table)) {
+      if (schemaCache.containsKey(table)) {
+        // Table already exists; noop
+        logger.debug("Skipping create of {} as it should already exist or appear very soon", table(table));
+        return;
+      }
+
+      TableInfo tableInfo = constructTableInfo(table, topic);
+      logger.info("Attempting to create {} with schema {}",
+          table(table), tableInfo.getDefinition().getSchema());
+      try {
+        bigQuery.create(tableInfo);
+        logger.debug("Successfully created {}", table(table));
+        schemaCache.put(table, tableInfo.getDefinition().getSchema());
+      } catch (BigQueryException e) {
+        if (e.getCode() == 409) {
+          logger.debug("Failed to create {} as it already exists (possibly created by another task)", table(table));
+          schemaCache.put(table, readTableSchema(table));
+        }
```

Comment:
Do something with the exception if the error code is not 409. Otherwise it would be silently swallowed.

Reply:
🤦‍♀️ this is why we have code review.